src/librustdoc/clean/mod.rs

```diff
@@ -862,8 +861,27 @@ fn clean_ty_generics<'tcx>(
         .collect::<Vec<_>>();
 
     for (param, mut bounds) in impl_trait {
+        let mut has_sized = false;
+        bounds.retain(|b| {
+            if b.is_sized_bound(cx) {
+                has_sized = true;
```

Comment:
Looking at this, I don't think it handles `~const Sized` (`TraitBoundModifier::MaybeConst`) correctly.
I pretty sure it results in `has_sized == false` making us add `+ ?Sized` to it.
Let me know if I should fix it here or in a follow-up (I guess there might be some more misuses of `is_sized_bound`).

Reply:
That answers the question then. ;)